### ISSUE TITLE
Bugs in batch normalization layer of nn.batchNorm1d()?

### ISSUE THREAD

**Issue by pandezhao** [pytorch/pytorch#9602]
There is something wrong with the 1 dimension batch normalization layer, the output of normalization layer is not related to the input, but only related to the BN's parameter.
For example:
`a = Variable(torch.Tensor([[1000,5000],[50,20]]))
b=torch.nn.BatchNorm1d(2)
c = b(a)
`
then we only got  
`c tensor([[ 0.8070,  0.8190], [-0.8070, -0.8190]])`
This value is not related with input, but only related to Batch Normalization layer's parameters.
`b.weight
Parameter containing:
tensor([ 0.8070,  0.8190])`

But when we tried three channel input, the output value will not related to the weight.
`a=torch.randn(3,2)
b=torch.randn(3,2)
bn=BatchNorm1d(2)
a1=bn(a)
b1=bn(b)
a1
tensor([[-0.9728,  0.0101],
        [ 0.8335,  0.2478],
        [ 0.1393, -0.2579]])

b1
tensor([[ 0.7941,  0.0316],
        [-0.9947, -0.2673],
        [ 0.2006,  0.2357]])
bn.weight
Parameter containing:
tensor([ 0.7440,  0.2066])
`
This bug only happens when input batch size equals 2. 
And my torch vision is 0.4.0.

**Comment by t-vi** [comment]
This is a mathematical property of batch norm - with 2 inputs, there aren't many options to force the first and second moment (you can have various signs and the effect of eps, but other than that it's +1 and -1).

**Comment by soumith** [comment]
BatchNorm1d is not BatchNorm. It's the `channel`-ized version of BatchNorm.

In BatchNorm1d, the channels are treated to be independent of each other.
In fact, the input to BatchNorm1d should be 3D, not 2D.

BatchNorm and BatchNorm*d are implemented exactly as described in the paper.
